File: TaskManager/LabFinal/LabFinal/AddCommentCommand.cs
namespace LabFinal;
    public class AddCommentCommand : ICommand
    {
        private readonly IssueRepository _repository;
        private readonly int _issueId;
        private readonly Comment _comment;
        private CommandLog _log;

        public AddCommentCommand(IssueRepository repository, int issueId, Comment comment)
        {
            _repository = repository;
            _issueId = issueId;
            _comment = comment;
        }

        public void Execute()
        {
            var issue = _repository.GetIssue(_issueId);
            issue.Comments.Add(_comment);
            _log = new CommandLog(CommandType.Update, $"Added comment to issue {_issueId}");
        }

        public void Undo()
        {
            var issue = _repository.GetIssue(_issueId);
            issue.Comments.Remove(_comment);
        }

        public CommandLog GetLog() => _log;
    }

File: TaskManager/TaskManagerSolution/src/AddCommentCommand.java
import java.util.List;

public class AddCommentCommand extends Command{
    private final IssueRepository repository;
    private final String issueId;
    private final Comment comment;

    public AddCommentCommand(IssueRepository repository, String issueId, Comment comment) {
        super("Added comment to issue", CommandType.UPDATE);
        this.repository = repository;
        this.issueId = issueId;
        this.comment = comment;
    }

    @Override
    public void execute() {
        Issue issue = repository.getIssue(issueId);
        issue.addComment(comment);
    }

    @Override
    public void undo() {
        Issue issue = repository.getIssue(issueId);
        List<Comment> comments = issue.getComments();
        if (!comments.isEmpty()) {
            comments.remove(comment);
        }
    }

}
